### PR TITLE
[workspace] fix: manage link does not do anything

### PR DESCRIPTION
**Pull request by Qxisylolo**
### Description
This pr addresses the issue of inactive "Manage" link in the data source selector when opening DevTools. Now, clicking the link will direct users to opensearch-dashboards/dataSources.

## Screenshot

https://github.com/user-attachments/assets/0d5572e6-078d-4fab-8709-ece38c990246

## Changelog
- fix: inactive manage link in the data source selector when opening DevTools

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by yubonluo — APPROVED** [review]
Nice fix

File: src/plugins/data_source_management/public/components/data_source_menu/types.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export interface DataSourceMenuProps<T = any> {
   uiSettings?: IUiSettingsClient;
   application?: ApplicationStart;
   setMenuMountPoint?: (menuMount: MountPoint | undefined) => void;
+  onChangeDevToolsModalVisible: () => void;
```
Comment (ruanyl) [inline src/plugins/data_source_management/public/components/data_source_menu/types.ts:43]: I suggest to rename `onChangeDevToolsModalVisible` -> `onManageDataSource` to make this interface sounds more generic, ideally data source plugin should not maintain logics of dev_tools plugins.

File: src/plugins/data_source_management/public/components/data_source_menu/types.ts [hunk 2]
```diff
@@ -81,6 +82,7 @@ export interface DataSourceSelectableConfig extends DataSourceBaseConfig {
   notifications: NotificationsStart;
   activeOption?: DataSourceOption[];
   dataSourceFilter?: (dataSource: SavedObject<DataSourceAttributes>) => boolean;
+  onChangeDevToolsModalVisible: () => void;
```
Comment (SuZhou-Joe) [inline src/plugins/data_source_management/public/components/data_source_menu/types.ts:85]: I thought we will just listen to application change while the modal is visible.

Comment (Qxisylolo) [reply src/plugins/data_source_management/public/components/data_source_menu/types.ts:85]: Thanks for the comment, the dev tool is just EuiOverlayMask and using core.application.currentAppId$ to monitor app ID only returns the app that opens the dev tool.

File: src/plugins/data_source_management/public/components/data_source_menu/types.ts (hunk 1 above)
Comment (Qxisylolo) [reply src/plugins/data_source_management/public/components/data_source_menu/types.ts]: Thanks for the advice, already changed the name~

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**